src/test/java/de/tum/cit/aet/artemis/programming/ProgrammingExerciseIntegrationTestService.java

```diff
@@ -2278,6 +2286,96 @@ void testReEvaluateAndUpdateProgrammingExercise_isNotSameGivenExerciseIdInReques
         request.put("/api/programming-exercises/" + programmingExercise.getId() + "/re-evaluate", programmingExerciseToBeConflicted, HttpStatus.CONFLICT);
     }
 
+    void test_redirectGetSolutionRepositoryFilesWithoutContent() throws Exception {
+        test_redirectGetSolutionRepositoryFilesWithoutContent((exercise, files) -> {
+            LocalRepository localRepository = new LocalRepository("main");
+            try {
+                hestiaUtilTestService.setupSolution(files, exercise, localRepository);
+            }
+            catch (Exception e) {
+                fail("Setup solution threw unexpected exception: " + e.getMessage());
+            }
+            return localRepository;
+        });
+    }
+
+    private void test_redirectGetSolutionRepositoryFilesWithoutContent(BiFunction<ProgrammingExercise, Map<String, String>, LocalRepository> setupRepositoryMock) throws Exception {
+        setupRepositoryMock.apply(programmingExercise, Map.ofEntries(Map.entry("A.java", "abc"), Map.entry("B.java", "cde"), Map.entry("C.java", "efg")));
+
+        var savedExercise = programmingExerciseRepository.findByIdWithTemplateAndSolutionParticipationElseThrow(programmingExercise.getId());
+
+        request.getWithForwardedUrl("/api/programming-exercises/" + programmingExercise.getId() + "/file-names", HttpStatus.OK,
+                "/api/repository/" + savedExercise.getSolutionParticipation().getId() + "/file-names");
+    }
+
+    void test_redirectGetTemplateRepositoryFilesWithContent() throws Exception {
+        test_redirectGetTemplateRepositoryFilesWithContent((exercise, files) -> {
+            LocalRepository localRepository = new LocalRepository("main");
+            try {
+                hestiaUtilTestService.setupTemplate(files, exercise, localRepository);
+            }
+            catch (Exception e) {
+                fail("Setup template threw unexpected exception: " + e.getMessage());
+            }
+            return localRepository;
+        });
+    }
+
+    private void test_redirectGetTemplateRepositoryFilesWithContent(BiFunction<ProgrammingExercise, Map<String, String>, LocalRepository> setupRepositoryMock) throws Exception {
+        setupRepositoryMock.apply(programmingExercise, Map.ofEntries(Map.entry("A.java", "abc"), Map.entry("B.java", "cde"), Map.entry("C.java", "efg")));
+
+        var savedExercise = programmingExerciseRepository.findByIdWithTemplateAndSolutionParticipationElseThrow(programmingExercise.getId());
+
+        request.getWithForwardedUrl("/api/programming-exercises/" + programmingExercise.getId() + "/template-files-content", HttpStatus.OK,
```

Comment:
could you maybe add a small comment explaining why we expect a forwarded URL.

Reply:
I changed it so it now checks for the file contents instead of the forward URL, which makes more sense here.